src/Spago/Command/Fetch.purs

```diff
@@ -615,3 +671,13 @@ cycleError :: PackageName -> TransitiveDepsResult -> TransitiveDepsResult
 cycleError dep result = result
   { errors { cycle = Set.insert dep result.errors.cycle } }
 
+-- | When we work with dependencies, we need to keep track of them separately
+-- | for core and test, because they need to be treated differently in some
+-- | contexts.
+type ByEnv a = { core :: a, test :: a }
```

Comment:
I'm not a fan of the name "env" for this, but I couldn't come up with a better one.

Reply:
Yeah. It feels like we could come up with a better one, but I can't think of anything as of now - it's good enough